BrowserKit/Sources/ToolbarKit/AddressToolbar/LocationView/LocationView.swift

```diff
@@ -271,7 +268,10 @@ final class LocationView: UIView, LocationTextFieldDelegate, ThemeApplicable, Ac
 
         // Start overlay mode & select text when in edit mode with a search term
         if shouldShowKeyboard == true && state.shouldSelectSearchTerm == true {
-            urlTextField.selectAll(nil)
+            DispatchQueue.main.async {
+                self.urlTextField.text = text
```

Comment:
By setting the text to the `urlTextField` before highlighting it, we ensure that if the url is too long to be visible in the `urlTextField`, we will see the beginning of the url if we rotate the phone from landscape to portrait.